portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,30 +27,50 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+
+  List<String> comments = new ArrayList<String>();
   private final Gson gson = new Gson();
 
   @Override
   public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
-        
-        // Initialise messages with three hard-coded values
-        List<String> messages = new ArrayList<String>();
-        messages.add("Lesser leather never weathered wetter weather better");
-        messages.add("Which witch is which?");
-        messages.add("Fred fed Ted bread, and Ted fed Fred bread");
-
-        // Convert to JSON format
-        String json = toJsonArray(messages);
-
-        // Send JSON as response
-        response.setContentType("application/json;");
-        response.getWriter().println(json);
-    }
 
-    /**
-     * Converts input to a JSON Array
-     */
-    private String toJsonArray(List<String> data) {
-        final String json = gson.toJson(data);
-        return json;
+    // Convert to JSON format
+    String json = toJsonArray(comments);
+
+    // Send JSON as response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+  * Converts input to a JSON Array
+  */
+  private String toJsonArray(List<String> data) {
+    final String json = gson.toJson(data);
+    return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "comment", "");
+
+    comments.add(text);
+
+    // Respond with the updated comments.
+    response.setContentType("text/html;");
+    response.getWriter().println(comments.toString());
   }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
```

Comment:
can you add @param javadoc tags for this function too.

Reply:
Done in the "added code review changes" commit